src/js/components/forms/TextInput.jsx

```diff
@@ -1,12 +1,35 @@
 import React, { PropTypes } from 'react';
+import { Icon } from '@jenkins-cd/react-material-icons';
+
 import { TextControl } from './TextControl';
 
+
+// wraps the Icon in a div to provide consistent cursor behavior
+function NestedIcon(props) {
+    return (
+        <div className={`TextInput-icon ${props.className}`}>
+            <Icon icon={props.icon} />
+        </div>
+    );
+}
+
+NestedIcon.propTypes = {
+    className: PropTypes.string,
+    icon: PropTypes.string,
+};
+
+
 export class TextInput extends React.Component {
 
     render() {
+        const classLeft = this.props.iconLeft ? 'u-icon-left' : '';
```

Comment:
Can you point me to the design for that? You're saying there are multiple icons on the left or right?